### PR TITLE
Slider: update styling to use Semantic Slots instead of the Fabric Palette colors.

### PR DESCRIPTION
**Pull request by aneeshack4**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Updated Slider's styling to use Semantic Slots instead of Fabric Palette colors and ran Screener locally and it passed with no errors.

It became pretty clear that the Semantic Slots are somewhat mis-aligned; the mappings to palette colors don't make full semantic sense for each slot so it was a bit hard to choose the right slot. I tried to explain my thought process in comments next to some of the slots. This is kind of an example first-try for moving our whole component library to Semantic Slots so that the rest of the team can see how to go about doing this for the remaining components and so we know what kinds of problems they'll face with the current mappings.

#### Focus areas to test

Check screener to make sure Slider's snaps didn't change in terms of the colors. Check example page on fabric docs website to make sure colors didn't change. 

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/9059)

### CODE REVIEW

**Comment by aftab-hassan** [comment]
@aneeshack4 I am taking a look at this.

I can confirm that size auditor did run, and size information for the baseline build IS present in the database (unlike an issue we've seen in the past when either master was broken or the Size auditor job did not run due to Azure Dev Ops SLA).

I compared the baseline and PR sizes from the database, and the only component that has changed in size is: 
`Slider` reduced in size by 5 bytes from 62115 to 62110.

PR
image

Baseline
image

I will take this up as a bug and investigate and report back here. Thanks for reaching out offline @aneeshack4

File: packages/office-ui-fabric-react/src/components/Slider/Slider.styles.ts [hunk 1]
```diff
@@ -33,7 +34,7 @@ export const getStyles = (props: ISliderStyleProps): ISliderStyles => {
   };
 
   const slideBoxInactiveSectionStyles = !props.disabled && {
-    backgroundColor: theme.palette.themeLight,
+    backgroundColor: theme.palette.themeLight, // There's no semantic slot with a value of hemeLight in the noneVariant, only strong & soft
```
Comment (phkuo) [inline packages/office-ui-fabric-react/src/components/Slider/Slider.styles.ts]: 1) Is this the right place to document these choices? Maybe.
2) The logic is backwards. It should not be "I need the themeLight slot, what slot has that color?" But rather "what semantic slot would make sense here?" If the answer is "none" then maybe we need to add a new semantic slot.
3) themeLight is spelled wrong.

File: packages/office-ui-fabric-react/src/components/Slider/Slider.styles.ts [hunk 2]
```diff
@@ -107,10 +108,10 @@ export const getStyles = (props: ISliderStyleProps): ISliderStyles => {
             {
               borderWidth: 2,
               borderStyle: 'solid',
-              borderColor: theme.palette.neutralSecondary,
+              borderColor: semanticColors.smallInputBorder, // Could use focusBorder also but better to keep with the input theme
```
Comment (phkuo) [inline packages/office-ui-fabric-react/src/components/Slider/Slider.styles.ts]: It's not "better" but correct to stay with the input theme. So focus border shouldn't even be considered.

File: packages/office-ui-fabric-react/src/components/Slider/Slider.styles.ts [hunk 3]
```diff
@@ -130,7 +131,7 @@ export const getStyles = (props: ISliderStyleProps): ISliderStyles => {
               transition: `left ${AnimationVariables.durationValue3} ${AnimationVariables.easeFunction1}`
             },
             props.disabled && {
-              borderColor: theme.palette.neutralTertiaryAlt,
+              borderColor: semanticColors.buttonBackgroundChecked, // None of the disabled semantic slots had a value of neturalTertiaryAlt
```
Comment (phkuo) [inline packages/office-ui-fabric-react/src/components/Slider/Slider.styles.ts]: See the first comment. This should be some disabled color slot. Even if we need to add a new one.

File: packages/office-ui-fabric-react/src/components/Slider/Slider.styles.ts [hunk 4]
```diff
@@ -193,7 +194,7 @@ export const getStyles = (props: ISliderStyleProps): ISliderStyles => {
     activeSection: [
       classNames.activeSection,
       {
-        background: theme.palette.neutralSecondary,
+        background: semanticColors.inputPlaceholderText,
```
Comment (phkuo) [inline packages/office-ui-fabric-react/src/components/Slider/Slider.styles.ts]: I don't think this is the right color to use here.

File: packages/office-ui-fabric-react/src/components/Slider/Slider.styles.ts [hunk 5]
```diff
@@ -204,7 +205,7 @@ export const getStyles = (props: ISliderStyleProps): ISliderStyles => {
         transition: `width ${AnimationVariables.durationValue3} ${AnimationVariables.easeFunction1}`
       },
       props.disabled && {
-        background: theme.palette.neutralTertiaryAlt,
+        background: semanticColors.buttonBackgroundChecked, // Only other semantic slots with value of neutralTertiaryAlt are menu slots
```
Comment (phkuo) [inline packages/office-ui-fabric-react/src/components/Slider/Slider.styles.ts]: See first comment again. Any disabled colors should pull from a disabled list.

File: packages/office-ui-fabric-react/src/components/Slider/Slider.styles.ts [hunk 6]
```diff
@@ -227,7 +228,8 @@ export const getStyles = (props: ISliderStyleProps): ISliderStyles => {
         transition: `width ${AnimationVariables.durationValue3} ${AnimationVariables.easeFunction1}`
       },
       props.disabled && {
-        background: theme.palette.neutralLight,
+        background: semanticColors.bodyDivider, // None of the disabled slots have value of neturalLight
```
Comment (phkuo) [inline packages/office-ui-fabric-react/src/components/Slider/Slider.styles.ts]: Same.

**Comment by aftab-hassan** [comment]
I re ran the PR once again, and it generates stats. Haven't seen this behavior before though. Will take a  look at logs and see what went wrong.

**Comment by aneeshack4** [comment]
Going to close this for now because (a) will have to do this off the new fabric 7 branch we're going to create to do all the conversion work off and (b) spoke with the designer and we both agree Slider is a tricky component because it uses color differently than other inputs and may even need new slots so holding off until that's figured out.

**Pull request closed without merging**